trestle/core/ssp_io.py

```diff
@@ -213,14 +215,53 @@ def get_control_response(self, control_id: str, level: int) -> str:
         """
         if not self._profile_path:
-            raise TrestleError('Cannot get control statement, set profile first.')
+            raise TrestleError('Cannot get control response, set profile first.')
 
-        writer = ControlIOWriter()
         control = self._catalog_interface.get_control(control_id)
-        group_id, group_title, _ = self._catalog_interface.get_group_info_by_control(control.id)
-        response_lines = writer.get_response(control)
+        control_impl_req = self._control_implemented_req(control_id)
 
-        tree = MarkdownNode.build_tree_from_markdown(response_lines)
+        md_writer = MDWriter(None)
+        if control_impl_req.statements:
+            for statement in control_impl_req.statements:
+                statement_id = statement.statement_id
+                label = statement_id
+                # look up label for this statement
+                if control.parts:
+                    found_label = self._catalog_interface.get_statement_label_if_exists(control_id, statement_id)
+                    if found_label:
+                        label = found_label
+
+                header = f'Implementation {label}'
+                response_per_component = {}
+                if statement.by_components:
+                    for idx, component in enumerate(statement.by_components):
+                        # look up component title
+                        subheader = component.uuid
+                        response = ''
+                        if self._ssp.system_implementation.components:
+                            for comp in self._ssp.system_implementation.components:
+                                title = comp.title
+                                if title == SSP_MAIN_COMP_NAME and idx == 0:
+                                    # special case dont print first "This system"
+                                    continue
+                                subheader = title
+                        if component.description:
+                            response = component.description
+
+                        if response or (not response and write_empty_responses):
+                            response_per_component[subheader] = response
+
+                if response_per_component or (not response_per_component and write_empty_responses):
+                    md_writer.new_header(level=level, title=header)
+                    for component_key in response_per_component:
+                        md_writer.new_header(level=level + 1, title=component_key)
+                        md_writer.set_indent_level(-1)
+                        md_writer.new_line(response_per_component[component_key])
+                        md_writer.set_indent_level(-1)
+
+        lines = md_writer.get_lines()
+
+        tree = MarkdownNode.build_tree_from_markdown(lines)
         tree.change_header_level_by(level)
```

Comment:
Should this be a change by level command or do we need a 'set to the top level in this block is X' command?

Reply:
`MarkdownNode` will adjust the markdown headers by the given level. By default all generated from SSP markdown text will start from `#` level